qa/tasks/cephfs/test_scrub_checks.py

```diff
@@ -30,22 +30,48 @@ def _resume_scrub(self, expected):
         self.assertEqual(res['return_code'], expected)
     def _get_scrub_status(self):
         return self.fs.rank_tell(["scrub", "status"])
-    def _check_task_status(self, expected_status):
-        task_status = self.fs.get_task_status("scrub status")
-        active = self.fs.get_active_names()
-        log.debug("current active={0}".format(active))
-        self.assertTrue(task_status[active[0]].startswith(expected_status))
+    def _check_task_status(self, expected_status, timo=120):
+        check = 0
+        while check < timo:
+            active = self.fs.get_active_names()
+            log.debug("current active={0}".format(active))
+            task_status = self.fs.get_task_status("scrub status")
+            try:
+                if task_status[active[0]].startswith(expected_status):
+                    return True
+            except KeyError:
+                pass
+            check += 1
+            time.sleep(1)
+        return False
+
+    def _check_task_status_na(self, timo=120):
+        check = 0
+        while check < timo:
+            active = self.fs.get_active_names()
+            log.debug("current active={0}".format(active))
+            task_status = self.fs.get_task_status("scrub status")
+            if not active[0] in task_status:
+                return True
+            check += 1
+            time.sleep(1)
+        return False
+
+    def create_scrub_data(self, test_dir):
+        self.mount_a.run_shell(["mkdir", test_dir])
+        for i in range(32):
+            dirname = "dir.{0}".format(i)
+            dirpath = os.path.join(test_dir, dirname)
+            self.mount_a.run_shell(["mkdir", dirpath])
+            for j in range(32):
+                filename = "file.{0}".format(j)
+                self.mount_a.write_n_mb(os.path.join(dirpath, filename), 1)
```

Comment:
Also check out `self.run_shell_payload`. I think we should start moving away from these awkward numerous calls to `run_shell` when a trivial monolithic shell script will do.

Reply:
ACK